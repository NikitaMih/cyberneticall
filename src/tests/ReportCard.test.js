import { render } from '@testing-library/react';
import ReportCard from '../components/reportCard';

const item = {
    symbol: 'SPAN', 
    latestPrice: '187', 
    avgTotalVolume: '674', 
    week52High: '376', 
    week52Low: '133'
};

describe('CompanyItem', () => {
    it('Should create ReportCard with params', () => {
        const view = render(
            <ReportCard {...item} />
        );
    
        expect(view).toMatchSnapshot();
    })
});
