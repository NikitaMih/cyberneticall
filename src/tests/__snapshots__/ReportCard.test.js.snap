// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyItem Should create CompanyItem with params 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <table>
          <tr>
            <th>
              Symbol
            </th>
            <tb>
              SPAN
            </tb>
          </tr>
          <tr>
            <th>
              Latest Price
            </th>
            <tb>
              187
            </tb>
          </tr>
          <tr>
            <th>
              Avg Total Volume
            </th>
            <tb>
              674
            </tb>
          </tr>
          <tr>
            <th>
              Week 52 High
            </th>
            <tb>
              376
            </tb>
          </tr>
          <tr>
            <th>
              Week 52 Low
            </th>
            <tb>
              133
            </tb>
          </tr>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <table>
        <tr>
          <th>
            Symbol
          </th>
          <tb>
            SPAN
          </tb>
        </tr>
        <tr>
          <th>
            Latest Price
          </th>
          <tb>
            187
          </tb>
        </tr>
        <tr>
          <th>
            Avg Total Volume
          </th>
          <tb>
            674
          </tb>
        </tr>
        <tr>
          <th>
            Week 52 High
          </th>
          <tb>
            376
          </tb>
        </tr>
        <tr>
          <th>
            Week 52 Low
          </th>
          <tb>
            133
          </tb>
        </tr>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CompanyItem Should create ReportCard with params 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <table>
          <tr>
            <th>
              Symbol
            </th>
            <tb>
              SPAN
            </tb>
          </tr>
          <tr>
            <th>
              Latest Price
            </th>
            <tb>
              187
            </tb>
          </tr>
          <tr>
            <th>
              Avg Total Volume
            </th>
            <tb>
              674
            </tb>
          </tr>
          <tr>
            <th>
              Week 52 High
            </th>
            <tb>
              376
            </tb>
          </tr>
          <tr>
            <th>
              Week 52 Low
            </th>
            <tb>
              133
            </tb>
          </tr>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <table>
        <tr>
          <th>
            Symbol
          </th>
          <tb>
            SPAN
          </tb>
        </tr>
        <tr>
          <th>
            Latest Price
          </th>
          <tb>
            187
          </tb>
        </tr>
        <tr>
          <th>
            Avg Total Volume
          </th>
          <tb>
            674
          </tb>
        </tr>
        <tr>
          <th>
            Week 52 High
          </th>
          <tb>
            376
          </tb>
        </tr>
        <tr>
          <th>
            Week 52 Low
          </th>
          <tb>
            133
          </tb>
        </tr>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
